### ISSUE TITLE
Error message is unclear on listing key from an unavailable rule name in servicebus namespace

### ISSUE THREAD

**Issue by evanlwj** [Azure/azure-cli#15726]
### **This is autogenerated. Please review and update as needed.**
When run az command to list keys on  an unavailable rule name in namespace, return error message "**TypeError: string indices must be integers**" is unclear, hard to understand the root cause. That would be good to reflect the actual error when accessing an unavailable rule name.

## Describe the bug
Error message is unclear to get unavailable rule name in namespace

**Command Name**
``
az servicebus namespace authorization-rule keys list --subscription ***  --name <rule name not existing> --namespace-name ** --resource-group ** --output json
``

**Errors:**
``
TypeError: string indices must be integers
``

## To Reproduce:
az servicebus namespace authorization-rule keys list --subscription ***  --name **\<rule name not existing\>** --namespace-name ** --resource-group ** --output json
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/knack/cli.py", line 215, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 654, in execute
    raise ex
  File "/usr/local/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 718, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/local/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 711, in _run_job
    six.reraise(*sys.exc_info())
  File "/usr/local/lib/python3.8/site-packages/six.py", line 703, in reraise
    raise value
  File "/usr/local/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 688, in _run_job
    result = cmd_copy(params)
  File "/usr/local/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 325, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/azure/cli/core/__init__.py", line 545, in default_command_handler
    return op(**command_args)
  File "/usr/local/lib/python3.8/site-packages/azure/mgmt/servicebus/operations/namespaces_operations.py", line 870, in list_keys
    raise models.ErrorResponseException(self._deserialize, response)
azure.mgmt.servicebus.models.error_response_py3.ErrorResponseException: Operation returned an invalid status code 'Not Found'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.8/site-packages/azure/cli/__main__.py", line 41, in <module>
    exit_code = cli_main(az_cli, sys.argv[1:])
  File "/usr/local/lib/python3.8/site-packages/azure/cli/__main__.py", line 30, in cli_main
    return cli.invoke(args)
  File "/usr/local/lib/python3.8/site-packages/knack/cli.py", line 226, in invoke
    exit_code = self.exception_handler(ex)
  File "/usr/local/lib/python3.8/site-packages/azure/cli/core/__init__.py", line 106, in exception_handler
    return handle_exception(ex)
  File "/usr/local/lib/python3.8/site-packages/azure/cli/core/util.py", line 96, in handle_exception
    error = response_dict['error']
**TypeError: string indices must be integers**
root@dcbb744c871f:/msftkube# az servicebus namespace authorization-rule keys list --subscription bb85ce66-75b5-49d6-a82a-28988b37^C76 --name ListenSendSharedAccessKey --namespace-name hyperdrive-fail-usnatwest --resource-group vienna-usn
atwest --output json

- _Put any pre-requisite steps here..._
- ``

## Environment Summary
```
Linux-4.19.128-microsoft-standard-x86_64-with-glibc2.29
Python 3.8.2
Installer: PIP

azure-cli 2.9.0 *

```

**Comment by yungezz** [comment]
add @houk-ms , typeError

**Comment by houk-ms** [comment]
@evanlwj Thanks for the feedbacks!
This issue should be resolved in the latest version, can you upgade CLI to 2.14.0 and try it again?

**Comment by v-Ajnava** [comment]
The issue is resolved, please upgrade Azure CLI  2.22.1.
hence closing the issue.
